qa/integration-tests/src/test/java/io/zeebe/broker/it/job/ActivateJobsTest.java

```diff
@@ -168,19 +173,55 @@ public void shouldActivateJobsOnPartitionsRoundRobin() {
     assertThat(activatedPartitionIds).containsOnlyElementsOf(partitionIds);
   }
 
+  @Test
+  public void shouldCompleteJobsIfBatchRecordIsTruncated()
+      throws IOException, InterruptedException {
+    // given
+    final int numJobs = 15;
+    final byte[] payloadBytes =
+        StreamUtil.read(
+            ActivateJobsTest.class.getResourceAsStream("/payloads/large_random_payload.json"));
+    final String payload = new String(payloadBytes, Charset.forName("UTF-8"));
+
+    createJobs(JOB_TYPE, numJobs, payload);
+
+    // when
+    final CountDownLatch latch = new CountDownLatch(numJobs);
+    client
+        .newWorker()
+        .jobType(JOB_TYPE)
+        .handler(
+            (client, job) -> {
+              client.newCompleteCommand(job.getKey()).send().join();
+              latch.countDown();
+            })
+        .name("worker")
+        .timeout(Duration.ofMinutes(2))
+        .bufferSize(10)
+        .open();
+    latch.await();
+
+    // then
+    assertThat(RecordingExporter.jobRecords(JobIntent.COMPLETED).limit(numJobs).count())
+        .isEqualTo(numJobs);
```

Comment:
We should ensure that truncation has happened. It can be that we increase the limit or change the JSON.